sdk/src/azure/core/az_credential_client_secret.c

```diff
@@ -88,6 +91,14 @@ AZ_NODISCARD az_result az_credential_client_secret_init(
     az_span client_id,
     az_span client_secret)
 {
+  _az_PRECONDITION_NOT_NULL(out_credential);
+  _az_PRECONDITION_VALID_SPAN(tenant_id, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client_id, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client_secret, 1, false);
+
+  static az_span const microsoft_aad_authority
+      = AZ_SPAN_LITERAL_FROM_STR("https://login.microsoftonline.com/");
```

Comment:
Can we define this once outside the method, similar to these:
https://github.com/Azure/azure-sdk-for-c/blob/355cfe5900f5186a4c94e1ad9ff7f0c5555b78be/sdk/src/azure/iot/az_iot_hub_client.c#L21-L22

Reply:
I don't think we should do it, if it is not used outside. If/when it is used outside, we absolutely should do it. If used locally, the more local definition is - the better.

Reply:
Does that end up creating the span repeatedly each time the caller calls this method, or only once, as if it was declared outside as a static?